static/scripts/rewards/render-transaction/tx-type.ts

```diff
@@ -1,16 +1,66 @@
-export type TxType = {
-  permit: {
-    permitted: {
-      token: string;
-      amount: string;
-    };
-    nonce: string;
-    deadline: string;
-  };
-  transferDetails: {
-    to: string;
-    requestedAmount: string;
-  };
-  owner: string;
-  signature: string;
-};
+import { Type as T, StaticDecode } from "@sinclair/typebox";
+import { BigNumber } from "ethers";
+
+const TBigNumber = T.Transform(T.Union([T.RegExp(/^\d+$/), T.Number()]))
+  .Decode(value => BigNumber.from(value))
+  .Encode(value => value.toString());
+
+const TNetworkId = T.Transform(T.Union([T.RegExp(/^0x\d+$/), T.Number()]))
+  .Decode(value => (typeof value === "number" ? "0x" + value.toString(16) : value))
+  .Encode(value => value);
+
+const TAddress = T.Transform(T.RegExp(/^0x[a-fA-F0-9]{40}$/))
+  .Decode(value => value.toLowerCase())
+  .Encode(value => value);
+
+const TSignature = T.Transform(T.RegExp(/^0x[a-fA-F0-9]+$/))
+  .Decode(value => value.toLowerCase())
+  .Encode(value => value);
+
+const Permit = T.Object({
+  type: T.Literal("permit"),
+  permit: T.Object({
+    permitted: T.Object({
+      token: TAddress,
+      amount: TBigNumber,
+    }),
+    nonce: TBigNumber,
+    deadline: TBigNumber,
+  }),
+  transferDetails: T.Object({
+    to: TAddress,
+    requestedAmount: TBigNumber,
+  }),
+  owner: TAddress,
+  signature: TSignature,
+  networkId: TNetworkId,
+});
+
+export type Permit = StaticDecode<typeof Permit>;
+
+const NftMint = T.Object({
+  type: T.Literal("nft-mint"),
+  request: T.Object({
+    beneficiary: TAddress,
+    deadline: TBigNumber,
+    keys: T.Array(T.String()),
+    nonce: TBigNumber,
+    values: T.Array(T.String()),
+  }),
+  nftMetadata: T.Object({
+    GITHUB_ORGANIZATION_NAME: T.String(),
+    GITHUB_REPOSITORY_NAME: T.String(),
+    GITHUB_ISSUE_ID: T.String(),
+    GITHUB_USERNAME: T.String(),
+    GITHUB_CONTRIBUTION_TYPE: T.String(),
+  }),
+  nftAddress: TAddress,
+  networkId: TNetworkId,
+  signature: TSignature,
+});
```

Comment:
I'm having type issues due to trying to process ERC20 and ERC721 permits in the UI code (the schemas don't line up at all.)

We should try and sync the properties as much as possible. I'm working on that now on my branch. I hope that it won't cause issues but I'm looking for input on constraints for the property names. 

From what I understand if we change the property names then the old permits will be incompatible.

Reply:
Because then you have erc721 specific properties like nftAddress, nftMetadata in erc20 type and the other way around so it's confusing

Reply:
We should add an extra property specifically for NFTs then. That way its very easy to deal with the generic `Permit` type, and then extend for `NftPermit`